### PR TITLE
node js task 2 done

### PR DESCRIPTION
**Pull request by MD-Liyaqath-Ali**

### CODE REVIEW

File: Node js Task2/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "express-starter",
+  "version": "1.0.0",
+  "description": "A basic Express Server to perform CRUD operations",
+  "main": "index.js",
+  "scripts": {
+    "start": "SET PORT=4000 && nodemon index.js",
```
Comment (leenabodhe) [inline Node js Task2/package.json:7]: Remove SET word

File: Node js Task2/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [
+    "express",
+    "data persistence"
+  ],
+  "author": "Md_Liyaqath_Ali@epam.com",
+  "license": "ISC",
+  "dependencies": {
```
Comment (leenabodhe) [inline Node js Task2/package.json:16]: you can add nodemon as a dev dependency

File: Node js Task2/RouteController.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+const UpdateUser = (req, res) => {
+    const id = req.params.id;
+    const data = req.body;
+    const position = users.findIndex((user) => user.id === id);
+    if (position != -1) {
+        const result = validateUser(data);
+        if (result.error == undefined) {
+            users[position] = data;
+            res.statusCode = 200;
+            res.send({ message: "Successfully Updated User's data" });
```
Comment (leenabodhe) [inline Node js Task2/RouteController.js:29]: Do not mix upper and lower case letters like above in sentence

File: Node js Task2/RouteController.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const id = req.params.id;
+    const position = users.findIndex((user) => user.id === id && user.isDeleted === false);
+    const userToBeDeleted = users[position];
+    if (position != -1) {
+        userToBeDeleted.isDeleted = true;
+        res.statusCode = 200;
+        res.send({ message: "Successfully Deleted a User's data" });
+    } else {
+        res.statusCode = 404;
+        res.send({ message: "Unable to find the user to delete OR Alread the user is set to be Deleted" });
```
Comment (leenabodhe) [inline Node js Task2/RouteController.js:60]: User not found message is enough other possibilities not required

File: Node js Task2/RouteController.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    res.send({message:"The Suggested Users",Users:suggestedUsers});
+};
+const deleteUser = (req, res) => {
+    const id = req.params.id;
+    const position = users.findIndex((user) => user.id === id && user.isDeleted === false);
+    const userToBeDeleted = users[position];
+    if (position != -1) {
+        userToBeDeleted.isDeleted = true;
+        res.statusCode = 200;
+        res.send({ message: "Successfully Deleted a User's data" });
```
Comment (leenabodhe) [inline Node js Task2/RouteController.js:57]: Please correct cases under sentence for all the messages rest looks good

**Review by leenabodhe — CHANGES_REQUESTED** [review]

File: Node js Task2/package.json (hunk 2 above)
Comment (MD-Liyaqath-Ali) [reply Node js Task2/package.json:16]: change made

File: Node js Task2/RouteController.js (hunk 4 above)
Comment (MD-Liyaqath-Ali) [reply Node js Task2/RouteController.js]: change made

File: Node js Task2/RouteController.js (hunk 5 above)
Comment (MD-Liyaqath-Ali) [reply Node js Task2/RouteController.js]: change made

**Review by leenabodhe — APPROVED** [review]

**Pull request merged**